src/main/java/io/github/treesitter/jtreesitter/internal/ChainedLibraryLookup.java

```diff
@@ -0,0 +1,30 @@
+package io.github.treesitter.jtreesitter.internal;
+
+import io.github.treesitter.jtreesitter.NativeLibraryLookup;
+import java.lang.foreign.Arena;
+import java.lang.foreign.Linker;
+import java.lang.foreign.SymbolLookup;
+import java.util.Optional;
+import java.util.ServiceLoader;
+
+final class ChainedLibraryLookup implements NativeLibraryLookup {
+    @Override
+    public SymbolLookup get(Arena arena) {
+        var serviceLoader = ServiceLoader.load(NativeLibraryLookup.class);
+        // NOTE: can't use _ because of palantir/palantir-java-format#934
+        SymbolLookup lookup = (name) -> Optional.empty();
+        for (var libraryLookup : serviceLoader) {
+            lookup = lookup.or(libraryLookup.get(arena));
+        }
+        return lookup.or(findLibrary(arena)).or(Linker.nativeLinker().defaultLookup());
+    }
+
+    private static SymbolLookup findLibrary(Arena arena) {
+        try {
+            var library = System.mapLibraryName("tree-sitter");
+            return SymbolLookup.libraryLookup(library, arena);
+        } catch (IllegalArgumentException e) {
+            return SymbolLookup.loaderLookup();
```

Comment:
As side note: This fallback with `SymbolLookup.loaderLookup()` probably still requires that the users call either `System#loadLibrary` or `System#load`, and even then only `System#loadLibrary` is affected by `java.library.path`.

So maybe it would be good to update the package Javadoc (which lists requirements and shows usage) to
- mention `System#loadLibrary` or `System#load`
(and maybe not even mention `java.library.path`)
- also mention the new `NativeLibraryLookup` interface, otherwise it might be easy to miss


What do you think @ObserverOfTime?